app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt

```diff
@@ -546,11 +516,37 @@ open class DefaultToolbarMenu(
     val deleteDataOnQuit = BrowserMenuImageText(
         label = context.getString(R.string.delete_browsing_data_on_quit_action),
         imageResource = R.drawable.ic_exit,
-        iconTintColorResource = primaryTextColor
+        iconTintColorResource = primaryTextColor()
     ) {
         onItemTapped.invoke(ToolbarMenu.Item.Quit)
     }
 
+    val syncedTabsItem = BrowserMenuImageText(
+        context.getString(R.string.synced_tabs),
+        R.drawable.ic_synced_tabs,
+        primaryTextColor()
+    ) {
+        onItemTapped.invoke(ToolbarMenu.Item.SyncedTabs)
+    }
+
+    private val syncSignInText = context.getString(R.string.sync_menu_sign_in)
+    private fun getSyncItemTitle(): String {
+        val accountExists = accountManager.authenticatedAccount() != null
+        return if (accountExists) {
```

Comment:
We might prefer the removed solution on lines -414 to -419 in order to avoid the ?:. What do you think?

return if (accountManager.authenticatedAccount() != null && accountManager.accountProfile()?.email != null) {
```suggestion
        return if (accountManager.authenticatedAccount() != null && accountManager.accountProfile()?.email != null) {
```

Reply:
That doesn't work, I need some sort of cast or fallback even if we check for null
<img width="982" alt="image" src="https://user-images.githubusercontent.com/43795363/114902072-158ea280-9ddb-11eb-94ee-69d1ea863096.png">

<img width="605" alt="image" src="https://user-images.githubusercontent.com/43795363/114902192-32c37100-9ddb-11eb-814e-fb5da6cea436.png">

Reply:
do you have a preference? `as String` would probably be my choice

Reply:
Ohh that's wild. I was just looking at this block from before.

```
    val accountManager = context.components.backgroundServices.accountManager
    val account = accountManager.authenticatedAccount()
    val syncItemTitle = if (account != null && accountManager.accountProfile()?.email != null) {
        context.getString(R.string.sync_signed_as, accountManager.accountProfile()?.email)
    } else {
        context.getString(R.string.sync_menu_sign_in)
    }
```
I think we should avoid manual casting. There's nothing particularly wrong with your previous solution. The only notable point I would make is that when I see a ?: after an if statement, it usually just means we should overload the if statement to avoid the ?:.

What if we set 
```
val email = accountManager.accountProfile()?.email
```
before the if statement?

Reply:
It will still need some sort of non-null assertion. Either 

```kotlin
val email = accountManager.accountProfile()?.email!!
``` 

or

```kotlin
val syncItemTitle = if (account != null && accountManager.accountProfile()?.email != null) {   
       email!!
}
```

Reply:
the old block was working because we had "Signed in as %1" as the string, and this PR is removing that